go/vt/vtctl/workflow/materializer.go

```diff
@@ -618,7 +618,7 @@ func primaryVindexesDiffer(ms *vtctldatapb.MaterializeSettings, source, target *
 		// If neither source nor target have any vindexes, treat the answer to
 		// the question as trivially false.
 		if len(sColumnVindexes) == 0 && len(tColumnVindexes) == 0 {
-			return true
+			return false
```

Comment:
This is an unrelated fix for https://github.com/vitessio/vitess/issues/16565 

This had no practical impact but was clearly wrong.

Reply:
In general I'm not in favor of mixing two separate bug fixes into one PR, mostly because different issues have different levels of severity, and you might not want to backport both into release branches.